lib/src/excel.dart

```diff
@@ -146,32 +153,39 @@ class ExcelDart {
           ? dailyMaxMicros
           : (Int64.parseInt(dailyMaxMicros) * 1e-6).toString();
 
-  /// Generates columns for budget unit, automation type,
-  /// active segments total budget, start date and end date from [budget]
-  static List<String> _generateBudgetColumns(InsertionOrder_Budget budget) {
+  /// Extracts active budget segments based on start and end dates.
+  static List<InsertionOrder_Budget_BudgetSegment> _extractActiveBudgetSegments(
+      List<InsertionOrder_Budget_BudgetSegment> budgetSegments) {
     final now = DateTime.now();
-    final activeSegments = budget.budgetSegments.where((segment) {
+    final activeSegments = budgetSegments.where((segment) {
       final startDate = segment.dateRange.startDate;
       final endDate = segment.dateRange.endDate;
       return now.isAfter(
               DateTime.utc(startDate.year, startDate.month, startDate.day)) &&
           now.isBefore(DateTime.utc(endDate.year, endDate.month, endDate.day));
     });
 
+    return activeSegments.toList();
+  }
+
+  static String _calculateTotalBudgetAmount(
+      List<InsertionOrder_Budget_BudgetSegment> activeSegments) {
     final totalBudgetMicros = activeSegments
         .map((segment) => Int64.parseInt(segment.budgetAmountMicros))
         .reduce((sumAmount, segmentAmount) => sumAmount + segmentAmount);
 
-    final startDate = activeSegments.first.dateRange.startDate;
-    final endDate = activeSegments.last.dateRange.endDate;
+    return (totalBudgetMicros.toDouble() * 1e-6).toString();
```

Comment:
is there accuracy loss from this? reporting api will also return in micros. so we should make converting micros into a util function and handle it carefully in one place

Reply:
That .toDouble() shouldn't be there. Sorry!

Reply:
That what it says in the documentation: On some platforms, inputs with large absolute values (i.e., > 2^52) may lose some of their low-order bits.

Reply:
Please see the latest commit for how this is changed.